src/main/kotlin/se/uulm/snowballr/backend/service/UserService.kt

```diff
@@ -148,4 +161,27 @@ class UserService(
 
         return tokens
     }
+
+    override suspend fun updateUser(request: UserOuterClass.User.Update): UserOuterClass.User {
+        val currentUser = userRepo.getUserById(GrpcContext.getUserIdFromContext())
```

Comment:
By adding this, we also check whether the requested user even exists, which is currently missing here.
```suggestion
        val currentUser = userRepo.getUserById(GrpcContext.getUserIdFromContext())
        val requestedUserId = parseUUID(request.user.id, "user")
        val requestedUser = userRepo.getUserById(requestedUserId)
```

Reply:
You are right, this test of existence I outsourced to the repository layer by checking whether the update worked. I am not sure, where it does belong and whether we really need to check this here?

Reply:
This should be resolved in the discussion above. But anyway, I thought about it again and in general, we should first make the access check and then any other check. So, please move the access check before the one for the duplicate email.

Reply:
I added your suggestion. Did I miss anything as I moved the checks around?

Reply:
Looks good.